src/AspNet.Security.OAuth.Odnoklassniki/OdnoklassnikiAuthenticationOptions.cs

```diff
@@ -0,0 +1,53 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+
+namespace AspNet.Security.OAuth.Odnoklassniki
+{
+    /// <summary>
+    /// Configuration options for <see cref="OdnoklassnikiAuthenticationMiddleware"/>.
+    /// </summary>
+    public class OdnoklassnikiAuthenticationOptions : OAuthOptions
+    {
+        /// <summary>
+        /// Initializes a new <see cref="OdnoklassnikiAuthenticationOptions"/>.
+        /// </summary>
+        public OdnoklassnikiAuthenticationOptions()
+        {
+            AuthenticationScheme = OdnoklassnikiAuthenticationDefaults.AuthenticationScheme;
+            DisplayName = OdnoklassnikiAuthenticationDefaults.DisplayName;
+            ClaimsIssuer = OdnoklassnikiAuthenticationDefaults.Issuer;
+
+            CallbackPath = new PathString(OdnoklassnikiAuthenticationDefaults.CallbackPath);
+
+            AuthorizationEndpoint = OdnoklassnikiAuthenticationDefaults.AuthorizationEndpoint;
+            TokenEndpoint = OdnoklassnikiAuthenticationDefaults.TokenEndpoint;
+            UserInformationEndpoint = OdnoklassnikiAuthenticationDefaults.UserInformationEndpoint;
+
+            Scope.Add("VALUABLE_ACCESS");
+            Scope.Add("GET_EMAIL");
+        }
+
+        public string ClientPublic { get; set; }
```

Comment:
`ClientPublic` is confusing as it makes me think that it's a flag used to determine whether the client app is a confidential or public app (which is absolutely not what it does).

Consider renaming it to `ApplicationKey` (it's the name I've used for the OpenID2 Steam provider: https://github.com/aspnet-contrib/AspNet.Security.OpenId.Providers/blob/dev/src/AspNet.Security.OpenId.Steam/SteamAuthenticationOptions.cs#L25)